src/components/Staking/ETHLPStaking/StakeForm/index.js

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { useForm } from 'react-hook-form';
+import PropTypes from 'prop-types';
+import { ethSelectors } from '../../../../redux/selectors';
+import { TextInput } from '../../../InputComponents';
+import Button from '../../../Button/Button';
+import { formatCustom } from '../../../../utils/walletHelpers';
+import { getTokenStakeOperations } from '../../../../api/ethereum';
+import styles from './styles.module.scss';
+
+function StakeForm({
+  account,
+  stakingToken,
+}) {
+  const {
+    handleSubmit,
+    register,
+    setValue,
+    watch,
+    setError,
+    formState: {
+      errors,
+      isSubmitting,
+      isSubmitSuccessful,
+    },
+  } = useForm({
+    mode: 'onChange',
+  });
+
+  const value = watch('stake', '');
+  const connected = useSelector(ethSelectors.selectorConnected);
+  const onSubmit = async ({ stake }) => {
+    try {
+      const signer = await connected.provider.getSigner(account);
+      const operations = getTokenStakeOperations(signer, stakingToken.address);
+      await operations.stake(stake);
+    } catch (e) {
+      setError('stake', {
+        message: 'Something went wrong',
+      });
+      // eslint-disable-next-line no-console
+      console.error(e);
+    } finally {
+      setValue('stake', '');
+    }
+  };
+
+  return (
+    <form
+      onSubmit={handleSubmit(onSubmit)}
+    >
+      <label className={styles.wrapper}>
+        Stake your
+        {' '}
+        {stakingToken.name}
```

Comment:
Stakingtoken.name will just be UniswapV2Pair if we read from the contract, i believe.


Ssoo its better not to read it and hardcode it to "ETH-LLD Uniswap v2 liquidity token"

Reply:
Renamed, also simplified the text on the "Stake all" button